frontend/src/components/UserOption.tsx

```diff
@@ -0,0 +1,11 @@
+export default function UserOption(props: any) {
+    
+    return (
```

Comment:
This is a sort of pseudocode example of what the code would look like for the required to work (i think??)
```js
<form onSubmit={sendDataToDBFunction}>
<label for="radio">Option</label>
<input 
    type="radio" 
    id="radio" 
    required 
/>
<input type="text" />
<button type="submit" />
</form>
```
everything is inside a form and you would have to select the radio button in order to submit the form, but the text field can be left empty since it doesn't have the required attribute. And the function that sends the data is in the form element itself. You don't need to attach an action to the actual button, since you are saying it is of type `submit`, it's already attached to the form.

Reply:
Also, not sure when the change in menu size is supposed to be triggered, but you could just add `onClick={setMenuOpen(true)}` or whatever where it's supposed to change. Or if it's something that can be clicked to toggle the state on and off you can do `onClick={(prev) => setMenuOpen(!prev)}` and that will make it so the menu will toggle between false and true on every click